src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotUseStackallocInLoopsTests.cs

```diff
@@ -63,6 +64,75 @@ unsafe class TestClass {
                 }");
         }
 
+        [Fact]
+        public async Task NoDiagnostics_StackallocInLoopButInsideALocalFunction()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp8,
+                TestCode = @"
+using System;
+class TestClass {
+    private static void StackAllocInLoopButInsideLocalFunction() {
+        while (true) {
+            XX();
+
+            static void XX()
+            {
+                Span<int> tmp = stackalloc int[10];
+                Console.WriteLine(tmp[0]);
+            }
+        }
+    }
+}"
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task NoDiagnostics_StackallocInLoopButInsideALambda()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp8,
+                TestCode = @"
+using System;
+class TestClass {
+    private static void StackallocInLoopButInsideALambda() {
+        while (true) {
+            Action a = () => {
```

Comment:
Can you also add a unit test for simple lambda expression syntax, i.e. say `Action<int> a = _ => Console.Write((stackalloc int[10]).Length);`

![image](https://user-images.githubusercontent.com/10605811/92500890-aa308280-f1b2-11ea-9684-d74cc2505c5a.png)